File: Week 6/solutions/A/A_abc.py
# input format: 'a' is a list of length 3, the 3 numbers in the first line
# 's' is a string, the 3 letters in the second line
# return the answer as a list of 3 numbers
def abc(a, s):
    b = []
    a.sort()
    for c in s:
        b.append(a[ord(c) - ord('A')])
    return b

a = list(map(int, input().split()))
s = input()
b = abc(a, s)
print(*b)

File: Week 6/solutions/A/A_abc.cpp
#include <bits/stdc++.h>
using namespace std;

// input format: 'a' is a vector of length 3, the 3 numbers in the first line
// return the answer as a vector of 3 numbers
vector<int> abc(vector<int> a, string s){
    vector<int> b;
    sort(a.begin(), a.end());   
    for(char c : s){
        b.push_back(a[c - 'A']);
    }
    return b;
}

int main(){
    vector<int> a(3);
    for(int i = 0; i < 3; i++){
        cin >> a[i];
    }
    string s;
    cin >> s;
    vector<int> b = abc(a, s);
    for(int i = 0; i < 3; i++){
        cout << b[i] << " ";
    }
    cout << '\n';
}